### ISSUE TITLE
Integral 1/(sin(x)**4 + cos(x)**4) can not be computed

### ISSUE THREAD

**Issue by sidhantnagpal** [sympy/sympy#14311]
```
>>> pprint(Integral(1/(sin(x)**4 + cos(x)**4), x))
⌠                     
⎮         1           
⎮ ───────────────── dx
⎮    4         4      
⎮ sin (x) + cos (x)   
⌡                     
```
Wolfram Alpha is able to integrate the [expression](http://www.wolframalpha.com/input/?i=Integral(1%2F(sin(x)**4+%2B+cos(x)**4),+x)).

**Comment by Roland-Sands** [comment]
Evaluating the integral requires `integrate(1/(sin(x)**4 + cos(x)**4), x)`
The computation hangs

**Comment by normalhuman** [comment]
Trigonometric power reduction seems to help (there was a suggestion to use it in heurisch):
```
>>> from sympy.simplify.fu import TRpower
>>> integrate(TRpower(1/(sin(x)**4 + cos(x)**4)), x) 
sqrt(2)*(atan(sqrt(2)*tan(2*x)/2) + pi*floor((2*x - pi/2)/pi))/2
```
The answer looks different. I haven't checked if it's equivalent to WA output.

**Comment by sidhantnagpal** [comment]
> The answer looks different. I haven't checked if it's equivalent to WA output.

Yes, I don't think that it is correct. It should not have term containing `floor` depending on `x`.

I solved it manually, by using `sin(x)**4 + cos(x)**4 = 1 - 2(sin(x)cos(x))**2 = 1 - 1/2 sin(2x)**2` and I am getting the integral as `1/√2 atan(tan(2x)/√2) + C`. 

I think, WA just expands this result for getting the output.

**Comment by jksuom** [comment]
The `floor` term is added to make the antiderivative continuous. Its derivative is zero. It is needed for computing definite integrals which cannot be computed by means of a periodic antiderivative.

**Comment by sidhantnagpal** [comment]
> It is needed for computing definite integrals which cannot be computed by means of a periodic antiderivative.

Oh, I didnt know that. Thanks for the information.

But shouldnt the coefficent of `atan` be `1/√2` since the `argument` inside is identical and derivative of second term is `0`.

**Comment by normalhuman** [comment]
WolframAlpha returns the function `g = (atan(sqrt(2)*tan(x) + 1) - atan(1 - sqrt(2) * tan(x)))/sqrt(2)` which is incorrect. This is clearly a bounded function while the true antiderivative of `1/(sin(x)**4 + cos(x)**4)` increases without bound as x->oo. Also, plotting g shows it's not continuous. 

The SymPy output is a continuous increasing function `f`. Numeric check shows it gives the correct integral values.
```
>>> N(f.subs(x, 10) - f.subs(x, 0))
14.0407579667151
>>> Integral(1/(sin(x)**4 + cos(x)**4), (x, 0, 10)).evalf()
14.0407579667151
```
(Incidentally, WolframAlpha agrees with this value for the definite integral.)

Advantage SymPy; but we need to tell it to use TRpower when appropriate.

**Comment by sidhantnagpal** [comment]
> Advantage SymPy; but we need to tell it to use TRpower when appropriate.

Yes, I think that would be advantageous.

But regarding the correctness (as I mentioned in my [previous comment](https://github.com/sympy/sympy/issues/14311#issuecomment-367957819)), derivative of SymPy output simplifies to `2/(sin(x)**4 + cos(x)**4)`, cross-checked via [WA](http://www.wolframalpha.com/input/?i=derivative+of+sqrt(2)*(atan(sqrt(2)*tan(2*x)%2F2) ), which should not be the case. 
Is something missing here?

**Comment by normalhuman** [comment]
I don't see it. The derivative has some stuff with Derivative(floor), which is generically zero. Discarding that and simplifying the rest, I get `2/(4*sin(x)**4 - 4*sin(x)**2 + 2)`. And `(2*sin(x)**4 - 2*sin(x)**2 + 1).equals(sin(x)**4 +cos(x)**4)` returns True.

**Comment by sidhantnagpal** [comment]
Ah, the braces. I misread the initial comment as, 

> `sqrt(2) atan(sqrt(2)*tan(2*x)/2)`

> Discarding that and simplifying the rest.

Yes, you are right.
